tests/unit/test_symmetric_crypto.cpp

```diff
@@ -532,3 +536,62 @@ TEST_F(SymmetricAuthenticatedCipherTest, throwsWhenWrongTagIsUsed) {
 
     ASSERT_THROW(decryptor->finish(), MoCOCrWException);
 }
+
+TEST_F(SymmetricAuthenticatedCipherTest, authenticatesAssociatedData) {
+    auto encryptor = AESCipherBuilder{SymmetricCipherMode::GCM,
+                                      SymmetricCipherKeySize::S_256,
+                                      _secretKey}.buildAuthenticatedEncryptor();
+
+    encryptor->setAssociatedData(_associatedData);
+    encryptor->update(_plaintext);
+    auto ciphertext = encryptor->finish();
+    auto tag = encryptor->getAuthTag();
+
+    auto decryptor = AESCipherBuilder{SymmetricCipherMode::GCM,
+                                      SymmetricCipherKeySize::S_256,
+                                      _secretKey}.setIV(encryptor->getIV()).buildAuthenticatedDecryptor();
+
+    decryptor->setAssociatedData(_associatedData);
+    decryptor->update(ciphertext);
+    decryptor->setAuthTag(tag);
+    auto decryptedText = decryptor->finish();
+    ASSERT_THAT(decryptedText, ::testing::ElementsAreArray(_plaintext));
+}
+
+TEST_F(SymmetricAuthenticatedCipherTest, throwsIfSetAssociatedDataCalledAfterUpdate) {
+    auto encryptor = AESCipherBuilder{SymmetricCipherMode::GCM,
+                                      SymmetricCipherKeySize::S_256,
+                                      _secretKey}.buildAuthenticatedEncryptor();
+
+    encryptor->update(_plaintext);
+    ASSERT_THROW(encryptor->setAssociatedData(_associatedData), MoCOCrWException);
+}
+
+TEST_F(SymmetricAuthenticatedCipherTest, throwsIfAssociatedDataMissingOrModified) {
+    auto encryptor = AESCipherBuilder{SymmetricCipherMode::GCM,
+                                      SymmetricCipherKeySize::S_256,
+                                      _secretKey}.buildAuthenticatedEncryptor();
+
+    encryptor->setAssociatedData(_associatedData);
+    encryptor->update(_plaintext);
+    auto ciphertext = encryptor->finish();
+    auto tag = encryptor->getAuthTag();
+
+    auto decryptor = AESCipherBuilder{SymmetricCipherMode::GCM,
+                                      SymmetricCipherKeySize::S_256,
+                                      _secretKey}.setIV(encryptor->getIV()).buildAuthenticatedDecryptor();
+    decryptor->setAuthTag(tag);
+    _associatedData[2] ^= _associatedData[2];
```

Comment:
XOR with itself is the same as writing `_associatedData[2] = 0x0`, isn't it? Did you mean to XOR with a static value?

Reply:
My bad. But that means that the other place in the code I mindlessly copied from also needs to be fixed. Will do it as a separate commit.